src/pcp/slabinfo/pcp-slabinfo.py

```diff
@@ -0,0 +1,206 @@
+        for name in names:
+            data = adjust_length(name) if len(name) < 25 else name
+            print("%s %s %s %s %8d %s %8d %s %8d %s %7d %s %9d %s %8d %s %8d"
+                                    % (timestamp, header_indentation,data, value_indentation, active_objects[name],
+                                       value_indentation,objsize[name],value_indentation, num_objs[name],
+                                       value_indentation, objperslab[name],value_indentation,pageperslab[name],
+                                       value_indentation, active_slabs[name], value_indentation, num_slabs[name]))
+
+    def print_report(self,group,timestamp,header_indentation,value_indentation,manager_slabinfo):
+        def __print_slab_status():
+            slabstatus = SlabStatUtil(manager_slabinfo)
+            if slabstatus.names():
+                try:
+                    self.__print_machine_info(group)
+                    self.__print_header(header_indentation, value_indentation)
+                    self.__print_values(timestamp, header_indentation, value_indentation, slabstatus)
+                except IndexError:
+                    print("Incorrect machine info due to some missing metrics")
+                return
+            else:
+                pass
+
+        if self.context != PM_CONTEXT_ARCHIVE and self.samples is None:
+            __print_slab_status()
+            sys.exit(0)
+        elif self.context == PM_CONTEXT_ARCHIVE and self.samples is None:
+            __print_slab_status()
+        elif self.samples >=1:
+            __print_slab_status()
+            self.samples-=1
+        else:
+            pass
+
+    def report(self, manager):
+        group = manager["sysinfo"]
+        self.samples=opts.pmGetOptionSamples()
+        t_s = group.contextCache.pmLocaltime(int(group.timestamp))
+        timestamp = time.strftime(SlabinfoOptions.timefmt, t_s.struct_time())
+        header_indentation = "        " if len(timestamp) < 9 else (len(timestamp) - 7) * " "
+        value_indentation = ((len(header_indentation) + 9) - len(timestamp)) * " "
+        self.print_report(group,timestamp,header_indentation,value_indentation,manager['slabinfo'])
+
+class SlabinfoOptions(pmapi.pmOptions):
+    timefmt = "%H:%M:%S"
+    def __init__(self):
+        pmapi.pmOptions.__init__(self, "a:s:Z:zV?")
+        self.pmSetLongOptionHeader("General options")
+        self.pmSetLongOptionHostZone()
+        self.pmSetLongOptionTimeZone()
+        self.pmSetLongOptionHelp()
+        self.pmSetLongOptionSamples()
+        self.pmSetLongOptionVersion()
+        self.samples=None
+        self.context=None
+
+if __name__ == '__main__':
+    try:
+        opts = SlabinfoOptions()
+        mngr = pmcc.MetricGroupManager.builder(opts,sys.argv)
+        opts.context=mngr.type
+        missing = mngr.checkMissingMetrics(ALL_METRICS)
+        if missing is not None:
+            sys.stderr.write(F"Error:Metric is {missing} missing\n")
```

Comment:
I think the grammar here is incorrect re "Metric is {...names-of-metrics...} missing",
perhaps something like the following would read better:
```
    sys.stderr.write("Error: some metrics are unavailable", missing)
```